src/storage/mod.rs

```diff
@@ -657,23 +664,31 @@ impl Storage {
                         ctx.get_isolation_level(),
                         !ctx.get_not_fill_cache(),
                     );
-                    snap_store
-                        .scanner(ScanMode::Forward, options.key_only, None, None)
-                        .and_then(|mut scanner| {
-                            let res = scanner.scan(start_key, limit);
-                            let statistics = scanner.get_statistics();
-                            thread_ctx.collect_scan_count(CMD, statistics);
-                            thread_ctx.collect_read_flow(ctx.get_region_id(), statistics);
-                            res
-                        })
-                        .map_err(Error::from)
-                        .map(|results| {
-                            thread_ctx.collect_key_reads(CMD, results.len() as u64);
-                            results
-                                .into_iter()
-                                .map(|x| x.map_err(Error::from))
-                                .collect()
-                        })
+
+                    let scan_mode = if !options.reverse_scan {
+                        ScanMode::Forward
+                    } else {
+                        ScanMode::Backward
+                    };
+
+                    let mut scanner = snap_store.scanner(scan_mode, options.key_only, None, None)?;
+                    let res = if !options.reverse_scan {
+                        scanner.scan(start_key, limit)
+                    } else {
+                        scanner.reverse_scan(start_key, limit)
+                    };
+
+                    let statistics = scanner.get_statistics();
```

Comment:
previous code only collect statistics when succeed

Reply:
No, previous code collect statistics after scanning, no matter if `res` is ok or not.